core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/CheckBoxEditorDirective.js

```diff
@@ -0,0 +1,291 @@
+            $scope.saveCustomizationArgs({
+              newValue: newCustomizationArgs
+            });
+            // The field needs to be initialized before the broadcast is sent.
+            $timeout(function() {
+              openChoiceEditor(newChoiceIndex);
+            });
+          };
+
+          $scope.isChoiceValid = function(index, newChoiceValue) {
+            var choiceNames = angular.copy(
+              $scope.getCustomizationArgs().choices.value);
+            choiceNames.splice(index, 1);
+
+            return (
+              Boolean(newChoiceValue) &&
+              choiceNames.indexOf(newChoiceValue) === -1);
+          };
+
+          $scope.saveChoice = function(index, newChoiceValue) {
+            if (!newChoiceValue) {
+              alertsService.addWarning('Cannot save an empty choice.');
+              return StatusObjectFactory.createFailure(
+                'Cannot save an empty choice'
+              );
+            }
+
+            var newCustomizationArgs = $scope.getCustomizationArgs();
+            var choiceNames = newCustomizationArgs.choices.value;
+
+            if (newChoiceValue === choiceNames[index]) {
+              // No change has been made.
+              return StatusObjectFactory.createFailure(
+                'No change has been made'
+              );
+            }
+
+            if (choiceNames.indexOf('newChoiceValue') !== -1) {
+              alertsService.addWarning(
+                'Cannot save: this duplicates an existing choice.');
+              return StatusObjectFactory.createFailure(
+                'This duplicates an existing choice'
+              );
+            }
+
+            newCustomizationArgs.choices.value[index] = newChoiceValue;
+            $scope.saveCustomizationArgs({
+              newValue: newCustomizationArgs
+            });
+          };
+
+          $scope.deleteChoice = function(value) {
+            var newCustomizationArgs = $scope.getCustomizationArgs();
+            if(newCustomizationArgs["choices"]["value"].length === 1){
+              throw Error(
+                'Cannot delete choice when there is only 1 choice remaining.');
+            }
+            else{
+              if(newCustomizationArgs["choices"]["value"].includes(value) ){
+                  var position = newCustomizationArgs["choices"]["value"].indexOf(value);
+                  newCustomizationArgs["choices"]["value"].splice(position, 1);
+                  newCustomizationArgs.maxAllowableSelectionCount.value--;
+                  newCustomizationArgs.minAllowableSelectionCount.value--;
+               }
+               var answerGroups = $scope.getAnswerGroups();
+               var newAnswerGroups = [];
+               newAnswerGroups.push(answerGroups[0]);
+               if(newAnswerGroups[0]["rules"][0]["inputs"]["x"].includes(value) ){
+                   var position = newAnswerGroups[0]["rules"][0]["inputs"]["x"].indexOf(value);
```

Comment:
@seanlip Just to check. Are we fine with brackets notation? Or dot notation is preferred?